src/common/keys.ts

```diff
@@ -0,0 +1,53 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { P0_PATH } from "../util";
+import * as fs from "fs/promises";
+import forge from "node-forge";
+import * as path from "path";
+
+export const PUBLIC_KEY_PATH = path.join(P0_PATH, "ssh", "id_rsa.pub");
+export const PRIVATE_KEY_PATH = path.join(P0_PATH, "ssh", "id_rsa");
+
+/**
+ * Search for a cached key pair, or create a new one if not found
+ */
+export const createKeyPair = async (): Promise<{
+  publicKey: string;
+  privateKey: string;
+}> => {
+  if (
+    (await fileExists(PUBLIC_KEY_PATH)) &&
+    (await fileExists(PRIVATE_KEY_PATH))
+  ) {
+    const publicKey = await fs.readFile(PUBLIC_KEY_PATH, "utf8");
+    const privateKey = await fs.readFile(PRIVATE_KEY_PATH, "utf8");
+
+    return { publicKey, privateKey };
+  } else {
+    const rsaKeyPair = forge.pki.rsa.generateKeyPair({ bits: 2048 });
```

Comment:
More a question/discussion point rather than something that I think should be changed in this PR: Was there a previous decision regarding selecting 2048-bit RSA keys for SSH? Ed25519 is a more modern standard for SSH keys which produces much smaller keys (so considerably easier to deal with than RSA keys for, say, passing in as a CLI argument or pasting into Slack) but still has security comparable to (if not perhaps a bit better than) 3072-bit RSA keys. It even has better performance, but wouldn't be noticeable for humans really. The only drawback is that Ed25519 is a newer standard so some very old SSH servers might not accept them, but anything reasonably modern should support it (OpenSSH added support for Ed25519 keys with version 6.5 in January 2014).

Reply:
We were using RSA's for now for the ephemeral generation, and in theory it also works for Ed25519, but since we're planning to move towards certificates the choice of algorithm has been deferred.